libyul/optimiser/DataFlowAnalyzer.cpp

```diff
@@ -271,7 +271,7 @@ void DataFlowAnalyzer::handleAssignment(set<YulString> const& _variables, Expres
 	auto const& referencedVariables = movableChecker.referencedVariables();
 	for (auto const& name: _variables)
 	{
-		m_state.references[name] = referencedVariables;
+		m_state.references.insert(name, referencedVariables);
```

Comment:
Shouldn't this clear at the same time?

Reply:
It indeed looks like it should, but the tests pass, which means that the coverage is maybe not great? Or could it be that since `clearValues` is called prior to this (line 253), it just so happens that the necessary entries (variables) are removed, but that's still a big assumption. I'll clear the values in the `insert` method in any case.